l3experimental/l3bitset/l3bitset.dtx

```diff
@@ -0,0 +1,637 @@
+% this index number is set and \texttt{0} otherwise.
+% \end{function}
+%
+% \begin{function}[EXP,added = 2020-12-13]
+%   {\bitset_to_bin:N, \bitset_to_bin:c}
+%   \begin{syntax}
+%     \cs{bitset_to_bin:N} \meta{bitset var}
+%   \end{syntax}
+% This leaves the current value of the bitset expressed as
+% a binary (string) number in the input stream.
+% If no bit has been set yet, the output is zero.
+% \end{function}
+% \begin{function}[EXP,added = 2020-12-13]
+%   {\bitset_to_arabic:N, \bitset_to_arabic:c}
+%   \begin{syntax}
+%     \cs{bitset_to_arabic:N} \meta{bitset var}
+%   \end{syntax}
+% This leaves the current value of the bitset expressed as
+% a decimal number in the input stream. If no bit has been set yet,
+% the output is zero. The function uses \cs{int_from_bin:n} if the largest
+% index that have been set or unset is smaller then 32, and a slower implementation
+% based on \cs{fp_eval} otherwise.
+% \end{function}
+%
+%
+% \begin{function}[added = 2020-12-13]
+%   {\bitset_show:N, \bitset_show:c}
+%   \begin{syntax}
+%     \cs{bitset_show:N} \meta{bitset var}
+%   \end{syntax}
+% Displays the binary and decimal value of the \meta{bitset var} on the terminal,
+% \end{function}
+%
+% \begin{function}[added = 2020-12-13]
+%   {\bitset_log:N, \bitset_log:c}
+%   \begin{syntax}
+%     \cs{bitset_log:N} \meta{bitset var}
+%   \end{syntax}
+% Writes the value of the \meta{bitset var} in the log file.
+% \end{function}
+%
+
+% \section{Syntactic sugar: Setting bits by keys}
+%
+% \begin{function}[added = 2020-12-19]
+%   {\bitset_addto_named_index:Nn}
+%   \begin{syntax}
+%     \cs{bitset_addto_named_index:Nn} \meta{bitset var}
+%      \{
+%         \meta{name1} |=| \meta{index1} |,|
+%         \meta{name2} |=| \meta{index2} |,| \ldots{}
+%      \}
+%   \end{syntax}
+% This adds the name--index pairs given in the second
+% argument to an internal, global property list. If the
+% property list didn't exist yet, it is created.
+% It is then possible to use the names (the keys of the property
+% to set and unset bits. It is not necessary to give every index
+% a name, and an index can have more than one name.
+% \end{function}
+%
+%  For example after this settings
+%
+%  \begin{verbatim}
+%  \bitset_new:N \l_pdfannot_F_bitset
+%  \bitset_addto_named_index:Nn \l_pdfannot_F_bitset
+%    {
+%      Invisible      = 1,
+%      Hidden         = 2,
+%      Print          = 3,
+%      NoZoom         = 4,
+%      NoRotate       = 5,
+%      NoView         = 6,
+%      ReadOnly       = 7,
+%      Locked         = 8,
+%      ToggleNoView   = 9,
+%      LockedContents = 10
+%    }
+%  \end{verbatim}
+%  it is possible to set bit 3 by using
+%  |\bitset_set_true_key:Nn \l_pdfannot_F_bitset {Print}|
+%  If a key is unknown a warning is issued.
+% \begin{function}[added = 2020-12-13]
+%   {\bitset_set_true_key:Nn, \bitset_set_true_key:cn,
+%    \bitset_gset_true_key:Nn,\bitset_gset_true_key:cn }
```

Comment:
Not sure about the use of "key" yet, but assuming we use "key" then I hink those would be all better called `\bitset_key_set_true:Nn` etc, i.e., keep `bitset_key` together